### PR TITLE
[DOC] GitHub spelling and links

### PR DESCRIPTION
**Pull request by oscargus**
## PR Summary

Two commits. One spelling GitHub correctly and one adding links for downloading C++-compilers and to example TeX distributions.

Download links is possibly primarily motivated by #24779 where finding/installing a C++-compiler is more tricky than on e.g. linux. The current links where primarily for proving that the stated version for C++-11 support is correct.

## PR Checklist

**Documentation and Tests**
- [ ] Has pytest style unit tests (and `pytest` passes)
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] New plotting related features are documented with examples.

**Release Notes**
- [ ] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [ ] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [ ] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

File: doc/devel/MEP/MEP19.rst [hunk 1]
```diff
@@ -47,7 +47,7 @@ has a number of shortcomings:
   outside of our control, causing unexpected delays at times that may
   not be convenient in our release schedule.
 
-On the plus side, Travis-CI's integration with github -- automatically
+On the plus side, Travis-CI's integration with GitHub -- automatically
```
[the remark this answers is not in the document]
Comment (oscargus) [reply doc/devel/MEP/MEP19.rst:50]: Yeah, I was doubting this one (and skipped the ones in e.g. release notes). Can revert.

[the remark this answers is not in the document]
Comment (ksunden) [reply doc/devel/MEP/MEP19.rst:50]: also... the mention of Travis is, well, outdated there

[the remark this answers is not in the document]
Comment (oscargus) [reply doc/devel/MEP/MEP19.rst:50]: Yeah, better to revert so we do not have to see it. ;-)

**Comment by tacaswell** [comment]
I have some concerns about adding the links to the compiler and latex installer pages.  For the most part people should be getting those through their system package manager not installing "by hand".  I worry the intersection of people who need use to give them the links to the compiler download pages and people who will be able to reliable get those to work (I have no faith I could quickly get a user-space install compiler to work) is small to non-existent so by adding this we are going to accidentally send people down very frustrating rabbit holes.

**Comment by oscargus** [comment]
From a Linux perspective: for sure. It is better to install from a package manager (I think most download pages actually says so as well).

From a Windows perspective: I am not so sure that people really know how to get a compiler up and running in a convenient way. I know on my previous laptop I had three Python versions and at least three compilers available, all in different shells, so getting a single shell where I could build C/C++ for Python was at least a bit challenging. Also, the Build Tools for Visual Studio are probably worth pointing out as they most likely are not that well known and probably the easiest way to get something up and running on Windows (mingw/cygwin clearly more complicated).

Maybe it is better to make specific Windows setup instructions then?

(For LaTeX it is not links to download pages, just two example packages, similar to Inkscape et al, which also should not be installed like that. Worth noting is that Graphviz on the following line actually points to the download page, which probably should be changed then.)

**Comment by jklymak** [comment]
@tacaswell, are you still semi-blocking on this?  We already link to compilers and latex packages, so are you advocating removing those, or not updating them?   Note this is the developer instructions, so we are allowed to assume some sophistication.

**Comment by tacaswell** [comment]
I still have some concerns but not enough to block the consensus, merging.

**Pull request merged**